providers/fips/self_test.c

```diff
@@ -141,14 +130,42 @@ void __TERM__cleanup(void) {
     cleanup();
 }
 
+#elif defined(__cplusplus)
```

Comment:
But as it stands this file will never be compiled with a c++ compiler?? This is a C file! If we have c++ code shouldn't we pull that out into a separate file?

Reply:
NIST seems to believe the languages are interchangable:

>Note7: The OS loaders of some operating systems do not provide a DEP mechanism for software libraries. In such cases the module shall utilize the available programming language capabilities to implement a DEP-like initialization as described in Note6 above. If the module is written in a procedural language, such as the C programming language, to avoid a complete rewrite in an object-oriented language, a judicious switch to a different compiler should be considered. For example, switching to a C++ compiler and placing the original C code inside extern “C” {} brackets, would enable static objects with the desired properties for the module.

Reply:
> placing the original C code inside extern “C” {} brackets, would enable static objects with the desired properties for the module.

You have not done this bit. Nor do I think you should in this file. This file should be pure C. If we need some C++ code as a fallback DEP then it should be in another file and would need to be compiled using a C++ compiler (which may not exist on all platforms).

Reply:
There should be no C++ fallback - that part of the IG simply defies logic. This is C code. This is not C++ code. And we will not introduce a C++ dependency. The languages are not interchangeable in this context.

Reply:
Removed.